projects/storefrontlib/cms-components/navigation/navigation/navigation-ui.component.ts

```diff
@@ -173,6 +180,29 @@ export class NavigationUIComponent implements OnInit, OnDestroy {
     event.stopPropagation();
   }
 
+  onSpace(event: UIEvent) {
```

Comment:
We should include unit tests here to test the functionality of these new methods. Possibly also e2e tests.
This is somewhat of a complex core component that does a lot of things and we really need to ensure nothing funny happens.

Reply:
I'm not super familiar with Jasmine thus not sure how to approach testing such a function. I have provided an e2e test though. :)

Reply:
I took a look at the e2e test and its spot on! Nice work

In regards to unit tests, we have some coverage checks to make sure that code is tested too. This usually blocks us from merging code that doesn't have enough testing (something like 90% of our code is required) which is why we need to keep on top of them. 

This is important if we are adding new methods to make sure they still behave exactly as we expect. E2Es do of course do a lot of this for us. But lib users will sometimes just use one or two methods from a component and expect them to behave in a certain way. That's why we need to have unit tests with the `'should do this' ... 'should call that' ... 'should NOT do this'` type scenarios too.

As in how to do this, there are plenty of examples in our `.spec.ts` files and it helps to find similar component tests to base off when possible. In our `KeyboardFocus` feature, we have plenty of keyboard manipulation going on to support a11y such of focus directives and tab traps. You can take a look at those spec files for what I think will be the most relevant for these cases (eg. the `TabFocusDirective` has tests for arrow key presses).

Reply:
I gave it a go and proposed some unit tests for the keyboard controls. Let me know if this is the right direction, thanks!